src/components/Transaction.tsx

```diff
@@ -1,31 +1,112 @@
+        source 
+        writable
+    }
     query testQuery($signature: Signature!) {
         transaction(signature: $signature) {
             blockTime
+            message {
+                accountKeys {
+                    ...TransactionMessageAccountKey
+                }
+                addressTableLookups {
+                    accountKey
+                    readableIndexes
+                    writableIndexes
+                }
+                header {
+                    numReadonlySignedAccounts 
+                    numReadonlyUnsignedAccounts 
+                    numRequiredSignatures 
+                }
+                recentBlockhash
+            }
+            meta {
+                fee
+                err
+                logMessages
+                postBalances
+                preBalances
+            }
+            signatures
             slot
+            version
         }
     }
 `;
 
-export default function Transaction() {
-    const [data, setData] = useState();
+/**
+ * Component GraphQL data.
+ */
+type Data = {
+    blockTime: string,
+    // Should Include more type to display info 
+    message: TransactionMessage,
+    // Should have a TransactionMetadata type in sdk
+    meta: {
+        fee: { __bigint: string} | undefined,
+        err: string | undefined,
+        logMessages: string | undefined,
+        postBalances: { __bigint: string} | undefined,
+        preBalances: { __bigint: string} | undefined 
+    }
+    signatures: [Signature],
+    slot: Slot,
+    version: String
+};
 
+interface Props {
+    signature: Signature 
+}
+
+export default function Transaction(props: Props) {
+    const [data, setData] = useState<Data>();
     useEffect(() => {
-        const fetchData = async () => {
-            let signature = '67rSZV97NzE4B4ZeFqULqWZcNEV2KwNfDLMzecJmBheZ4sWhudqGAzypoBCKfeLkKtDQBGnkwgdrrFM8ZMaS3pkk' as Signature;
-            const response = (await gql(source, { signature }));
-            console.log(response)
-            setData(response.account);
+        const fetchData = async() => {
+            const { signature } = props;
+            const { transaction } = (await gql(source, { signature })) as { transaction: Data };
+            setData(transaction);
         };
         fetchData();
     },[]);
-    return (<></>);
+    return (
+        <>
+            <p>Transaction with signatue: {props.signature}</p>
+            {data && 
+                <>
+                    <p>BlockTime: {data.blockTime.toString()}</p>
+
+                    <div>
+                        <h3>Transaction Meta:</h3>
+                        <p>Error: {data.meta.err}</p>
+                        <p>Fee: {data.meta.fee?.__bigint?.toString()}</p>
+                        <p>Log Messages: {data.meta.logMessages}</p>
+                        <p>Post Balances: {data.meta.postBalances?.__bigint?.toString()}</p>
+                        <p>Pre Balances : {data.meta.preBalances?.__bigint?.toString()}</p>
```

Comment:
If any of these often come up undefined, like `meta.err`, we can just omit them from the rendering for now.